src/main/java/strategy/PurchaseHandler.java

```
@@ -0,0 +1,15 @@
+package strategy;
+
+import db.Storage;
+import model.FruitTransaction;
+import service.OperationHandler;
+
+public class PurchaseHandler implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction fruitTransaction) {
+        String fruitName = fruitTransaction.getFruit();
+        int purchasedQuantity = fruitTransaction.getQuantity();
+        int oldQuantity = Storage.fruits.get(fruitName);
```

lets throw an exception if there is not enough fruits in the storage